Tests/LibLine/TestMetrics.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2025, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibTest/TestCase.h>
+
+#include <LibLine/Editor.h>
+#include <LibUnicode/Forward.h>
+
+TEST_CASE(count_ascii_glyphs_u8)
+{
+    constexpr auto string = "Hello, World!"sv; // length in bytes: 13, code points: 13, glyphs: 13
+    auto metrics = Line::Editor::actual_rendered_string_metrics(string);
+    EXPECT_EQ(metrics.grapheme_breaks.size(), 13u);
+    EXPECT_EQ(metrics.line_metrics.size(), 1u);
+    EXPECT_EQ(metrics.line_metrics[0].length, 13u);
+    EXPECT_EQ(metrics.line_metrics[0].visible_length, 13u);
+}
+
+TEST_CASE(count_ascii_glyphs_u32)
+{
+    constexpr u32 string[] = { 'H', 'e', 'l', 'l', 'o', ',', ' ', 'W', 'o', 'r', 'l', 'd', '!' }; // length in code points: 13, glyphs: 13
+    auto metrics = Line::Editor::actual_rendered_string_metrics(Utf32View(string));
+    EXPECT_EQ(metrics.grapheme_breaks.size(), 13u);
+    EXPECT_EQ(metrics.line_metrics.size(), 1u);
+    EXPECT_EQ(metrics.line_metrics[0].length, 13u);
+    EXPECT_EQ(metrics.line_metrics[0].visible_length, 13u);
+}
+
+TEST_CASE(count_simple_multibyte_glyphs_u8)
+{
+    constexpr auto string = "Héllo, Wörld!"sv; // length in bytes: 15, code points: 13, glyphs: 13
+    auto metrics = Line::Editor::actual_rendered_string_metrics(string);
+    EXPECT_EQ(metrics.grapheme_breaks.size(), 13u);
+    EXPECT_EQ(metrics.line_metrics.size(), 1u);
+    EXPECT_EQ(metrics.line_metrics[0].length, 13u);
+    EXPECT_EQ(metrics.line_metrics[0].visible_length, 13u);
+}
+
+TEST_CASE(count_simple_multibyte_glyphs_u32)
+{
+    constexpr u32 string[] = { 'H', 0xe9, 'l', 'l', 'o', ',', ' ', 'W', 0xf6, 'r', 'l', 'd', '!' }; // length in code points: 13, glyphs: 13
+    auto metrics = Line::Editor::actual_rendered_string_metrics(Utf32View(string));
+    EXPECT_EQ(metrics.grapheme_breaks.size(), 13u);
+    EXPECT_EQ(metrics.line_metrics.size(), 1u);
+    EXPECT_EQ(metrics.line_metrics[0].length, 13u);
+    EXPECT_EQ(metrics.line_metrics[0].visible_length, 13u);
+}
+
+TEST_CASE(count_multi_codepoint_glyphs_u8)
+{
+    if (!Unicode::is_data_available())
```

Comment:
We generally have taken the approach of just not caring about whether UCD downloads were disabled for running tests. There are lots of tests that will fail in that case (e.g. test-js), which is fine IMO.